app/controllers/accounts/pipelines_controller.rb

```diff
@@ -122,29 +122,33 @@ def export
   end
 
   def bulk_action
-    @pipeline = Pipeline.find(params[:pipeline_id])
+    @pipeline = current_user.account.pipelines.find(params[:pipeline_id])
+    @stage = current_user.account.stages.find(params[:stage_id])
     @event = Event.new
   end
 
   def create_bulk_action
     @deals = current_user.account.deals.where(stage_id: params['stage_id'], status: 'open')
-    time_start = DateTime.current
-
+    if params['event']['send_now'] == 'true'
+      time_start = DateTime.current
+    else
+      time_start = params['event']['scheduled_at'].to_time
+    end
     @result = @deals.each do | deal |
-      time_start = time_start + rand(5..15).seconds
-
-      Event.create(
-        deal: deal,
-        contact: deal.contact,
-        from_me: true, account: current_user.account,
-        kind: 'wpp_connect_message',
-        app_id: params['event']['app_id'],
-        app_type: params['event']['app_type'],
-        content: params['event']['content'],
-        custom_attributes: {'wpp_connect_message_to': deal.contact.phone},
-        scheduled_at: time_start
-      )
+      if params['event']['kind'] == 'chatwoot_message'
+        if params['event']['send_now'] == 'true'
+          time_start = time_start + rand(5..15).seconds
+          params['event']['send_now'] = 'false'
+        elsif !time_start.nil?
+          time_start = time_start + rand(5..15).seconds
+        end
+      end
```

Comment:
```suggestion
      time_start = time_start + rand(5..15).seconds

      if params['event']['kind'] == 'chatwoot_message'
        if params['event']['send_now'] == 'true'
          params['event']['send_now'] = 'false'        
        end
      end
```

Reply:
Essa alteração da erro pq se o time_start for nil, ele irá fazer o ` time_start = time_start + rand(5..15).seconds` e retornará erro. Por isso que eu adicionei o `elsif !time_start.nil?` na linha 143.